src/generator/generator_visitor.ts

```diff
@@ -215,6 +218,11 @@ class GeneratorVisitor extends Visitor {
     protected visitVariableDeclarator(node: VariableDeclarator) {
         if (node.init !== null) {
             this.visit(node.init);
+
+            if (isArrayExpression(node.init)) {
```

Comment:
This should be done in the `visitArrayExpression` method.

Reply:
That's not possible, because I need the variable name to create the local array at the correct memory position.